src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -379,346 +64,97 @@
+  @Mutation mutateFileCorrectionDialog!: (x: boolean) => void
+  @Mutation mutateLoadCorrectionDialog!: (x: boolean) => void
 
-  /** The Temporary Registration Number string. */
-  get tempRegNumber (): string {
-    return sessionStorage.getItem('TEMP_REG_NUMBER')
+  /** Whether this entity is a business (and not a temporary registration). */
+  get isBusiness (): boolean {
+    return !!sessionStorage.getItem('BUSINESS_ID')
   }
 
-  /** The Business ID string. */
-  get businessId (): string {
-    return sessionStorage.getItem('BUSINESS_ID')
+  /** Whether this entity is a temporary registration (and not a business). */
+  get isTemporaryRegistration (): boolean {
+    return !!sessionStorage.getItem('TEMP_REG_NUMBER')
   }
 
   /** Whether to show the history panel. */
   get showHistoryPanel () {
-    return (this.historyItems.length > 0)
-  }
-
-  /** Returns whether the action button is visible or not. */
-  protected displayAction (filing): string {
-    return filing.availableOnPaperOnly || filing.isTypeStaff || filing.documentsLink
-  }
-
-  /** Loads list of filings from the API into History Items array. */
-  private loadData (): void {
-    this.historyItems = []
-
-    // create 'history items' list from 'filings' array from API
-    for (const filing of this.getFilings) {
-      // safety check for required fields
-      if (!filing.name || !filing.displayName || !filing.effectiveDate || !filing.submittedDate || !filing.status) {
-        // eslint-disable-next-line no-console
-        console.log('Invalid filing =', filing)
-        continue
-      }
-
-      this.loadFiling(filing)
+    return (this.getFilings.length > 0)
+  }
+
+  /** Returns the name of the sub-component to use for the specified filing. */
+  protected is (filing: ApiFilingIF): string {
+    switch (true) {
+      case filing.availableOnPaperOnly: return 'paper-filing' // must come first
+      case EnumUtilities.isTypeAlteration(filing): return 'alteration-filing'
+      case EnumUtilities.isTypeChangeOfAddress(filing): return 'change-of-address'
+      case EnumUtilities.isTypeConsentContinuationOut(filing): return 'consent-continuation-out'
+      case EnumUtilities.isTypeDissolutionVoluntary(filing): return 'dissolution-voluntary'
+      case EnumUtilities.isTypeIncorporationApplication(filing): return 'incorporation-application'
+      case EnumUtilities.isTypeRestorationLimited(filing): return 'limited-restoration'
+      case EnumUtilities.isTypeRestorationConversion(filing): return 'limited-restoration-conversion'
+      case EnumUtilities.isTypeRestorationExtension(filing): return 'limited-restoration-extension'
+      case EnumUtilities.isTypeRegistration(filing): return 'registration-filing'
+      case EnumUtilities.isTypeStaff(filing): return 'staff-filing'
+      default: return 'default-filing'
     }
-
-    // report number of items back to parent (dashboard)
-    this.$emit('history-count', this.historyItems.length)
```

Comment:
This is no longer needed. Instead `getFilings.length` is used (in only 2 places).

Reply:
Beautiful!